src/solc_0.8/claims/MultiGiveaway/MultiGiveaway.sol

```diff
@@ -134,11 +175,26 @@ contract MultiGiveaway is AccessControl, ClaimERC1155ERC721ERC20, ERC2771Handler
         return ERC1155_BATCH_RECEIVED;
     }
 
+    /// @dev Returns the real sender of meta-transactions
+    /// @return sender sender of the meta-tx
     function _msgSender() internal view override(Context, ERC2771Handler) returns (address sender) {
         return ERC2771Handler._msgSender();
     }
 
+    /// @dev Returns the real data of meta-transactions
+    /// @return calldata data of the meta-tx
     function _msgData() internal view override(Context, ERC2771Handler) returns (bytes calldata) {
         return ERC2771Handler._msgData();
     }
+
+    /// @dev Prevent the renunciation of the DEFAULT_ADMIN_ROLE
+    /// @param role the role to revoke
+    /// @param account the address being revoked
+    function _revokeRole(bytes32 role, address account) internal virtual override {
+        if (role == DEFAULT_ADMIN_ROLE) {
+            revert("MULTIGIVEAWAY_PREVENT_REVOKE");
```

Comment:
Is this the intention, to prevent _all_ addresses that are given DEFAULT_ADMIN_ROLE from having it revoked?

Reply:
yes, QSP-2